### PR TITLE
VPC: VPE scoped development

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**
This PR contains 2 changes
1. VPE development changes (test cases are not tested properly, since it is scoped branch development)
2. Instance resource changes for resolving compilation errors.  Since, the instance resource code is not compatible with scoped version of vpc-go-sdk. So, I have modified the instance resource code. But did not test it as it will not be used by instance resource.

### CODE REVIEW

File: ibm/data_source_ibm_is_endpoint_gateway_ips.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	return &schema.Resource{
+		Read:     dataSourceIBMISEndpointGatewayIPsRead,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+
+			isEndpointGatewayID: {
+				Type:     schema.TypeString,
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_endpoint_gateway_ips.go:21]: sdk supports only id input

**Comment by Anil-CM** [comment]
@KumarGanesanIBM As we discussed already, there is no need to change the is_instance* resources. Please revert the changes with is_instance_* resource changes.

File: ibm/resource_ibm_is_endpoint_gateway_ip.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+				Description: "Endpoint gateway IP auto delete",
+			},
+			isEndpointGatewayIPAddress: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway IP address",
+			},
+			isEndpointGatewayIPTarget: {
+				Type:        schema.TypeMap,
+				Computed:    true,
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_is_endpoint_gateway_ip.go]: changed to TypeList.

File: ibm/resource_ibm_is_endpoint_gateway.go [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+				ForceNew:     false,
+				ValidateFunc: validateISName,
+				Description:  "Endpoint gateway name",
+			},
+			isEndpointGatewayResourceType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Endpoint gateway resource type",
+			},
+			isEndpointGatewayResourceGroup: {
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_is_endpoint_gateway.go]: changed to resource_group.

File: ibm/resource_ibm_is_endpoint_gateway.go [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+						isEndpointGatewayIPsResourceType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The VPC Resource Type",
+						},
+					},
+				},
+			},
+			isEndpointGatewayTarget: {
+				Type:        schema.TypeMap,
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_is_endpoint_gateway.go]: changed to TypeList

File: ibm/resource_ibm_is_endpoint_gateway.go [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+						},
+						isEndpointGatewayTargetResourceType: {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The target resource type",
+						},
+					},
+				},
+			},
+			isEndpointGatewayVpc: {
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_is_endpoint_gateway.go]: done

File: ibm/data_source_ibm_is_endpoint_gateway.go [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Endpoint gateway name",
+						},
+						isEndpointGatewayResourceType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Endpoint gateway resource type",
+						},
+						isEndpointGatewayResourceGroup: {
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_endpoint_gateway.go]: done

File: ibm/data_source_ibm_is_endpoint_gateway.go [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+}
+
+func dataSourceIBMISEndpointGatewaysRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := myvpcClient(meta)
+	if err != nil {
+		return err
+	}
+	opt := sess.NewListEndpointGatewaysOptions()
+	result, response, err := sess.ListEndpointGateways(opt)
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_endpoint_gateway.go]: done

File: ibm/data_source_ibm_is_endpoint_gateway_ips.go [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+							Description: "Endpoint gateway IP auto delete",
+						},
+						isEndpointGatewayIPAddress: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Endpoint gateway IP address",
+						},
+						isEndpointGatewayIPTarget: {
+							Type:        schema.TypeMap,
+							Computed:    true,
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_endpoint_gateway_ips.go]: changed to TypeList.

File: ibm/data_source_ibm_is_endpoint_gateway_ips.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func dataSourceIBMISEndpointGatewayIPsRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := myvpcClient(meta)
+	if err != nil {
+		return err
+	}
+	gatewayID := d.Get(isEndpointGatewayID).(string)
+	opt := sess.NewListEndpointGatewayIpsOptions(gatewayID)
+	result, response, err := sess.ListEndpointGatewayIps(opt)
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_endpoint_gateway_ips.go]: done

File: ibm/provider.go [hunk 10]
```diff
@@ -217,6 +217,8 @@ func Provider() terraform.ResourceProvider {
 			"ibm_is_instance_group":                 dataSourceIBMISInstanceGroup(),
 			"ibm_is_instance_group_manager":         dataSourceIBMISInstanceGroupManager(),
 			"ibm_is_instance_group_manager_policy":  dataSourceIBMISInstanceGroupManagerPolicies(),
+			"ibm_is_endpoint_gateways":              dataSourceIBMISEndpointGateways(),
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/provider.go]: done

**Comment by hkantare** [comment]
I think you need add docs

File: ibm/data_source_ibm_is_virtual_endpoint_gateways.go [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+const (
+	ibmISVirtualEndpointGateways = "ibm_is_virtual_endpoint_gateways"
+	isVirtualEndpointGateways    = "virtual_endpoint_gateways"
+)
+
+func dataSourceIBMISEndpointGateways() *schema.Resource {
+	return &schema.Resource{
+		Read:     dataSourceIBMISEndpointGatewaysRead,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_virtual_endpoint_gateways.go]: removed

File: ibm/data_source_ibm_is_virtual_endpoint_gateway_ips.go [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.ibm.com/ibmcloud/vpc-go-sdk/vpcv1"
+	vpcscoped "github.ibm.com/ibmcloud/vpc-go-sdk/vpcv1"
+)
+
+func dataSourceIBMISEndpointGatewayIPs() *schema.Resource {
+	return &schema.Resource{
+		Read:     dataSourceIBMISEndpointGatewayIPsRead,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
```
[the remark this answers is not in the document]
Comment (KumarGanesanIBM) [reply ibm/data_source_ibm_is_virtual_endpoint_gateway_ips.go]: removed

**Pull request merged**